app/services/hyrax/flexible_schema_validator_service.rb

```diff
@@ -14,31 +14,72 @@ class FlexibleSchemaValidatorService
 
     attr_reader :profile, :schema, :schemer, :errors
 
+    # Initializes a new FlexibleSchemaValidatorService.
+    #
+    # @param profile [Hash] the flexible metadata profile to validate
+    # @param schema [Pathname, String] the JSON schema to validate against.
+    #   Defaults to {DEFAULT_SCHEMA}.
+    # @return [void]
     def initialize(profile:, schema: default_schema)
       @profile = profile
       @schema = schema
       @schemer = JSONSchemer.schema(schema)
       @errors = []
     end
 
+    # Execute all validation routines and populate {#errors} with any
+    # issues discovered.
+    #
+    # @return [void]
     def validate!
       validate_required_classes
       validate_class_availability
+      validate_available_on_classes_defined
       validate_schema
       validate_label_prop
+      validate_enabled_work_types
       CoreMetadataValidator.new(profile: profile, errors: @errors).validate!
     end
 
+    # The default JSON schema used when no custom schema is provided.
+    #
+    # @return [Pathname]
     def default_schema
       DEFAULT_SCHEMA
     end
 
+    # Classes that MUST be present in every flexible metadata profile.
+    #
+    # @return [Array<String>]
     def required_classes
       REQUIRED_CLASSES
     end
 
     private
 
+    # Validates that the profile's work types correspond to the site's
+    # enabled work types, adding human-readable error messages for any
+    # discrepancies.
+    #
+    # @return [void]
+    def validate_enabled_work_types
```

Comment:
This will soon be replaced by the work in
- https://github.com/notch8/hykuup_knapsack/issues/429